docs: add admin api (#203)
GreptimeTeam/docs: docs/developer-guide/meta/admin-api.md

# Admin API

The Admin API provides a simple way to view cluster information, including meta health detection, meta leader query, database metadata query, and datanode heartbeat detection. 

> The Admin API is an HTTP service that provides a set of RESTful APIs that can be called through HTTP requests. The Admin API is designed to be simple, easy to use and safe.

Reviewer: ```suggestion
The Admin API is an HTTP service that provides a set of RESTful APIs that can be called through HTTP requests. The Admin API is simple, user-friendly and safe.
```

> Available API:

Reviewer: ```suggestion
Available APIs:
```

- /health
- /leader
- /catalogs
- /schemas
- /tables
- /heartbeat

All these APIs are under the parent resource `/admin`.

> The following sections assume your meta instance is running on localhost port 3002.

Reviewer: ```suggestion
In the following sections, we assume that your meta instance is running on localhost port 3002.
```

## /health HTTP endpoint  

> The `/health` endpoint accepts Get HTTP requests. Use this endpoint to check the health of your meta instance.

Reviewer 2: ```suggestion
The `/health` endpoint accepts GET HTTP requests. Use this endpoint to check the health of your meta instance.
```

> The `/health` endpoint accepts Get HTTP requests. Use this endpoint to check the health of your meta instance.

Reviewer: ```suggestion
The `/health` endpoint accepts Get HTTP requests and you can use this endpoint to check the health of your meta instance.
```

### Definition

```bash
curl -X GET http://localhost:3002/admin/health
```

### Examples

#### Request

```bash
curl -X GET http://localhost:3002/admin/health
```

#### Response

```json
OK
```

## /leader HTTP endpoint

> The `/leader` endpoint accepts Get HTTP requests. Use this endpoint to query the leader's addr of your meta instance.

Reviewer 2: ```suggestion
The `/leader` endpoint accepts GET HTTP requests. Use this endpoint to query the leader's addr of your meta instance.
```

### Definition

```bash
curl -X GET http://localhost:3002/admin/leader
```

### Examples

#### Request

```bash
curl -X GET http://localhost:3002/admin/leader
```

#### Response

```json
127.0.0.1:3002
```

## /catalogs HTTP endpoint

> The `/catalogs` endpoint accepts Get HTTP requests. Use this endpoint to query the all the catalog's name.

Reviewer 2: ```suggestion
The `/catalogs` endpoint accepts GET HTTP requests. Use this endpoint to query the all the catalog's name.
```

Reviewer 3: @jun0315 Please take a look at this comment. And please help to fix the conflict when you have time.

### Definition

```bash
curl -X GET http://localhost:3002/admin/catalogs
```

### Examples

#### Request

```bash
curl -X GET http://localhost:3002/admin/catalogs
```

#### Response

```json
["greptime"]
```

## /schemas HTTP endpoint

> The `/schemas` endpoint accepts Get HTTP requests. Use this endpoint to query all schemas of a specific catalog.

Reviewer: ```suggestion
The `/schemas` endpoint accepts Get HTTP requests and you can use this endpoint to query all schemas of a specific catalog.
```

### Definition

```bash
curl -X GET http://localhost:3002/admin/schemas
```

| Query String Parameter | Type   | Optional/Required | Definition               |
|:-----------------------|:-------|:------------------|:-------------------------|
| catalog_name           | String | Required          | The name of the catalog. |

### Examples

#### Request

```bash
curl -X GET 'http://localhost:3002/admin/schemas?catalog_name=greptime'
```

#### Response

```json
["public"]
```

## /tables HTTP endpoint

> The `/tables` endpoint accepts Get HTTP requests. Use this endpoint to query all tables of a specific catalog and schema.

Reviewer: ```suggestion
The `/tables` endpoint accepts Get HTTP requests and you can use this endpoint to query all tables of a specific catalog and schema.
```

> Therefore, specify `catalog_name` and `schema_name` in the path is required

Reviewer: ```suggestion
Therefore, it is required to specify `catalog_name` and `schema_name` in the path.
```

### Definition

```bash
curl -X GET http://localhost:3002/admin/tables
```

| Query String Parameter | Type   | Optional/Required | Definition               |
|:-----------------------|:-------|:------------------|:-------------------------|
| catalog_name           | String | Required          | The name of the catalog. |
| schema_name            | String | Required          | The name of the schema.  |

### Examples

#### Request

```bash
curl -X GET `http://localhost:3002/admin/tables?catalog_name=greptime&schema_name=public`
```

#### Response

```json
["dist_table"]
```

## /heartbeat HTTP endpoint

> The `/heartbeat` endpoint accepts Get HTTP requests. Use this endpoint to query the heartbeat of all datanodes.

Reviewer: ```suggestion
The `/heartbeat` endpoint accepts Get HTTP requests and you can use this endpoint to query the heartbeat of all datanodes.
```

> We can also query the heartbeat data of the datanode of the specified addr.

Reviewer: ```suggestion
You can also query the heartbeat data of the datanode for a specified `addr`, however, specifying `addr` in the path is optional.
```

> specify `addr` in the path is optional. 

Reviewer: ```suggestion

```
